src/data/sqlite_converter.py

```diff
@@ -0,0 +1,389 @@
+from dataconverter import DataConverter
+import pandas as pd
+from sqlalchemy import create_engine
+import sqlalchemy
+import time
+from multiprocessing import Pool
+from utils import I3Extractor, extract_retro
+import os
+from icecube import icetray, dataio
+import numpy as np
+from utils import load_geospatial_data
+import sqlite3
+from tqdm import tqdm
+from utils import I3Extractor
+
+
+def apply_event_no(extraction, event_no_list, event_counter):
+    out = pd.DataFrame(extraction.values()).T
+    out.columns = extraction.keys()
+    out['event_no'] = event_no_list[event_counter]
+    return out
+
+def isi3(file):
+    if 'gcd' in file.lower():
+        return False
+    elif 'geo' in file.lower():
+        return False
+    else:
+        return True
+def has_extension(file, extensions):
+    check = 0
+    for extension in extensions:
+        if extension in file:
+            check +=1
+    if check >0:
+        return True
+    else:
+        return False
+
+def fetch_temps(path):
+    out = []
+    files = os.listdir(path)
+    for file in files:
+        if '.db' in file:
+            out.append(file)
+    return out
+
+def walk_directory(dir, extensions, gcd_rescue):
+    files_list = []
+    GCD_list   = []
+    root,folders,root_files = next(os.walk(dir))
+    gcds_root = []
+    gcd_root = None
+    i3files_root = []
+    for file in root_files:
+        if has_extension(file, extensions):
+            if isi3(file):
+                i3files_root.append(os.path.join(root,file))
+            else:
+                gcd_root = os.path.join(root,file)
+                gcds_root.append(os.path.join(root,file))
+    if gcd_root == None:
+        gcd_root = gcd_rescue
+    for k in range(len(i3files_root) - len(gcds_root)):
+        gcds_root.append(gcd_root)
+    files_list.extend(i3files_root)
+    GCD_list.extend(gcds_root)
+    for folder in folders:
+        sub_root, sub_folders, sub_folder_files = next(os.walk(os.path.join(root,folder)))
+        gcds_folder = []
+        gcd_folder = None
+        i3files_folder = []
+        for sub_folder_file in sub_folder_files:
+            if has_extension(sub_folder_file, extensions):
+                if isi3(sub_folder_file):
+                    i3files_folder.append(os.path.join(sub_root,sub_folder_file))
+                else:
+                    gcd_folder = os.path.join(sub_root,sub_folder_file)
+                    gcds_folder.append(os.path.join(sub_root,sub_folder_file))
+        if gcd_folder == None:
+            gcd_folder = gcd_rescue
+        for k in range(len(i3files_folder) - len(gcds_folder)):
+            gcds_folder.append(gcd_folder)
+        files_list.extend(i3files_folder)
+        GCD_list.extend(gcds_folder)
```

Comment:
This seems like a somewhat involved function, and that might be necessary and if I please ignore this comment -- but at a first glance I would expect a combination of the `glob` and `re` packages and filtering to get most of the job done?

Reply:
Maybe. 

Lets zoom about this some time

Rasmus